app/services/move_participant.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class MoveParticipant < BaseService
+  attr_reader :participant_profile, :school
+
+  def initialize(participant_profile:, school:)
+    @participant_profile = participant_profile
+    @school = school
+  end
+
+  def call
+    cohort = participant_profile.school_cohort.cohort
+    school_cohort = school.school_cohorts.find_by!(cohort: cohort)
+    return if school_cohort.blank?
+
+    ActiveRecord::Base.transaction do
+      participant_profile.teacher_profile.update!(school: school)
+      participant_profile.update!(school_cohort: school_cohort,
+                                  sparsity_uplift: school.sparsity_uplift?(cohort.start_year),
+                                  pupil_premium_uplift: school.pupil_premium_uplift?(cohort.start_year))
```

Comment:
@blackrat do we want this?

Reply:
If it's to fix an error, yes. For people changing schools normally, no, they should have the original school values. The uplift payment is a "reward" for recruitment. If they change schools, that's a later thing that doesn't effect it.

Reply:
If it's an error, yes. If it's a genuine teacher changing schools, then no. It's a business rule based on where they were when they were recruited as opposed to where they actually are now.